### PR TITLE
Revert "Remove unnecessary truncating"

### PR DESCRIPTION
**Pull request by Androbin**
Reverts #286

### CODE REVIEW

**Comment by Androbin** [comment]
We might still want to de-duplicate the template.

**Comment by shardulc** [comment]
The `truncate` function can be moved to `tools.py`, and then perhaps used by `more.py` and `irc.py`.

**Comment by Androbin** [comment]
@shardulc Do we really want `more.py` and `irc.py` to truncate messages?
I mean, `break_up_fn` -> yes, `truncate` -> no.

**Comment by wei2912** [comment]
Truncation should be avoided. But they're good functions to have anyways.

**Comment by wei2912** [comment]
As in, it should not be used to reduce the length of messages for all modules, as we would lose information.

**Comment by wei2912** [comment]
How does the truncation code work, btw?

**Comment by Androbin** [comment]
* I have moved the function as desired
* `truncate` cuts off at the rightmost space below the limit

File: tools.py [hunk 1]
```diff
@@ -19,6 +19,12 @@
    'User-Agent': 'Mozilla/5.0' + '(X11; U; Linux i686)' + 'Gecko/20071127 Firefox/2.0.0.11'
 }
 
+def truncate(non_trunc, trunc):
+    while len(non_trunc + trunc) > MAX_MSG_LEN:
```
Comment (shardulc) [inline tools.py]: Move `MAX_MSG_LEN` from `git.py` to here too. Also, eventually we need to fix it by making it a global for all modules, so leave a TODO above it.

File: tools.py [hunk 2]
```diff
@@ -19,6 +19,23 @@
    'User-Agent': 'Mozilla/5.0' + '(X11; U; Linux i686)' + 'Gecko/20071127 Firefox/2.0.0.11'
 }
 
+# maximum message length (see msg() in irc.py)
+# overriden if MAX_MSG_LEN exists in the config
+# TODO: make this a global for all modules
+MAX_MSG_LEN = 430
+
+def setup(self):
```
Comment (shardulc) [inline tools.py]: As `tools.py` isn't a module, this function will never be called. (Which is why that global `MAX_MSG_LEN` thing is important.)

Comment (Androbin) [reply tools.py]: From where exactly?

Comment (Androbin) [reply tools.py]: Also, maybe merge #284 first and let me rebase, otherwise this will get tricky.

**Comment by shardulc** [comment]
@Androbin please update this, minimal changes now.

**Comment by Androbin** [comment]
@shardulc Could you point out from where exactly to call the `setup` function?

**Comment by Androbin** [comment]
Since you said "`tools.py` isn't a module", I figured I'd just call it right above their initialization.
Besides, I can't really figure where we would call this from `irc.py`.

**Pull request merged**